### ISSUE TITLE
How to push view on SWRevealViewController Using UInavigationViewController

### ISSUE THREAD

**Issue by ghost** [John-Lluch/SWRevealViewController#728]
actually i need to push ViewController On swrevealViewControlller . When i am trying to do so It is giving parent as nil.
if i am presenting it is working fine.  i am new to swift Please help me .
presentingVC

let storyboard = UIStoryboard(name: "Main", bundle: nil)
        let SWRController = storyboard.instantiateViewController(withIdentifier:"SWRevealViewController")
        switch (userType.loader) {  
        case .Agent:
            let destination = storyboard.instantiateViewController(withIdentifier: "Agent")
            (SWRController as? SWRevealViewController)?.frontViewController = destination
            self.present(SWRController, animated: true, completion:nil)
            break
pushing VC

 let destination = storyboard.instantiateViewController(withIdentifier: "Agent") as!UINavigationController
            let agentVC = destination.topViewController as! AgentVC
            (SWRController as? SWRevealViewController)?.frontViewController = agentVC
            self.navigationController?.pushViewController(agentVC, animated: true)
            
            break
it gives parent nil please helpme

**Comment by iDevelopper** [comment]
How is your storyboard?

**Comment by ghost** [comment]
i didn't get  you!  i am new to swift can u explain briefly

**Comment by iDevelopper** [comment]
Just show a screenshot of your Main.storyboard.

**Comment by ghost** [comment]
<img width="846" alt="screen shot 2017-08-31 at 5 27 59 pm" src="https://user-images.githubusercontent.com/30462604/29922205-3761ce18-8e72-11e7-9176-5a686f4b414a.png">

**Comment by iDevelopper** [comment]
Sorry can you upload the main.storyborad (you have to compress it in a zip) itself because this is too small in the screenshot.

**Comment by iDevelopper** [comment]
You have to push the SWRController not the agentVC

**Comment by ghost** [comment]
kk Thank you So much sir

**Comment by ghost** [comment]
[Main.storyboard.zip](https://github.com/John-Lluch/SWRevealViewController/files/1270294/Main.storyboard.zip)
if i am pushing i am not  getting menu option
<img width="305" alt="screen shot 2017-09-01 at 6 42 15 pm" src="https://user-images.githubusercontent.com/30462604/29971294-5eb131b0-8f45-11e7-959c-ba5872376140.png">


if  i am  presenting I am able to see menu button options




<img width="305" alt="screen shot 2017-09-01 at 6 45 53 pm" src="https://user-images.githubusercontent.com/30462604/29971447-e557136a-8f45-11e7-859b-0739688b4b75.png">
please help me sir

**Comment by iDevelopper** [comment]
Try to embed the controller in a navigation controller.

**Comment by iDevelopper** [comment]
And you don't have a sw_front segue. It should be HomeVC, no?

**Comment by iDevelopper** [comment]
[SWIssue728.zip](https://github.com/John-Lluch/SWRevealViewController/files/1270903/SWIssue728.zip)

**Comment by ghost** [comment]
Thank You so much !!!!!!!It working Now.

**Comment by ghost** [comment]
hi sir could you please help me in writing generic handler class URL Request. Thanks in advance.

**Comment by iDevelopper** [comment]
Can you elaborate? What kind of request? Did you start writing something?

**Comment by ghost** [comment]
I need to write a ServerRequestHandler which would be generic and which i could reuse in every other ViewControllers of my application. right now am replicating the same set of logic in every set of ViewControllers which is not object oriented.Can you please help me to write that Generic class. BTW i am using URLRequest and URLSession to do so.

**Comment by ghost** [comment]
func purchaseRequest()
    {
    
    let urlString = String(format:"http://mandi.connect2community.in/c4t/ledger?queryId=QUERY_DAY_BOOK&args=community_id:7f6a033f-6ae8-4e27-bb58-66238fc3be84,last_update_time:0,account_id:7f6a033f-6ae8-4e27-bb58-66238fc3be84^2017-2018^1,type:SALES,records:20")
    
        let encodedurl = urlString.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
    
    let url = URL(string:encodedurl!)
    
    
    var request = URLRequest(url: url! as URL)
    
    let result = UserDefaults.standard.value(forKey:"session_id")
    let ss : String = result as! String
    request.httpMethod = "GET"
    
    request.addValue("application/x-www-form-urlencoded", forHTTPHeaderField: "Content-Type")
    request.addValue(("session_id="+ss), forHTTPHeaderField: "Cookie")
    
    request.timeoutInterval = 3
    
    let   task = URLSession.shared.dataTask(with: request as URLRequest!, completionHandler: { (data, response, error) in
    
    
    if error != nil{
    print(error?.localizedDescription as Any)
    return
    }
    
    
    do {
    let json = try JSONSerialization.jsonObject(with: data!, options: .mutableContainers) as? NSDictionary
    
    print(json ?? "empty from server ")
    
    
    if let parseJSON = json {
    print(parseJSON)
    
    let  success  = Int(json?["errCode"]! as! Double)
    
    
    if  let resultArray = parseJSON["resource"] as?[[String:AnyObject]]{
    
        for (index,_) in resultArray.enumerated()
        {
    
    if let vocherno = resultArray[index]["voucher_no"]as? String                    {
    
        print(vocherno)
        if let eventDate = resultArray[index]["event_date"]as? Int                     {
                
                let event_date = Constants.getDateStringforMessageVC(dateConveriosn:Double(eventDate))
                
            
            print(event_date)
            if let particular = resultArray[index]["particular"]as? String                     {
                    
                    
                    
                
                print(particular)
                if let accountName = resultArray[index]["account_name"]as? String                     {
                        
                        
                        
                    
                        print(accountName)
            
            
                    
                    if let amount = resultArray[index]["dr_amount"]as? Int                     {
                        
                    
                    
self.purchaseresult.append(Purchase(vocher: vocherno, eventdate: event_date, particular: particular, accountname: accountName, amount: amount))
    
    print(amount)
            

                
                
            

            
            
        

    
    
                    }}}}}}
    
    }
    if (success   != 0) {
    return
    }
    else{
    DispatchQueue.main.async {
        self.daybooktableview.reloadData()
    }
    }}
    }
    
    catch let error as NSError {
    print(error)
    }
    })
    
    task.resume()
    
    
    }

**Comment by iDevelopper** [comment]
Ok, can you give me an URL that will work in a browser? And return a JSON.

**Comment by ghost** [comment]
These are two demo api's . Please use the first login to get a session id and use that session id in the get request of the second type. Please do open them up in postman rest api client - chrome extension. 

FYI, i attach the screenshots of the postman . 

And pass this in body of the login api request . as ** resource ** as eydlbWFpbF9pZCc6JzkwNjY2NjkzOTMnLCdwYXNzd29yZCc6J21hbmRpJ30=

<img width="1280" alt="screen shot 2017-09-03 at 4 56 20 pm" src="https://user-images.githubusercontent.com/30462604/30002529-faf1d244-90c8-11e7-9fc7-d41840817579.png">
<img width="1280" alt="screen shot 2017-09-03 at 4 56 35 pm" src="https://user-images.githubusercontent.com/30462604/30002530-faf452bc-90c8-11e7-87c3-2d71a3d29d05.png">
<img width="1280" alt="screen shot 2017-09-03 at 4 56 43 pm" src="https://user-images.githubusercontent.com/30462604/30002531-faf59a32-90c8-11e7-8fa8-8d022f8c3d03.png">






[Demo.postman_collection.zip](https://github.com/John-Lluch/SWRevealViewController/files/1272879/Demo.postman_collection.zip)

**Comment by iDevelopper** [comment]
Nothing in the file Demo.postman_collection.zip

**Comment by ghost** [comment]
You need to open with the Postman application. which is a chrome extension or an app

https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop?hl=en-GB&utm_source=chrome-ntp-launcher

or  https://www.getpostman.com/

import that file in postman

FYI, please check the screenshot attached.

<img width="1280" alt="screen shot 2017-09-03 at 6 11 28 pm" src="https://user-images.githubusercontent.com/30462604/30002936-72945cfe-90d3-11e7-8dac-f4586d58f66c.png">

**Comment by iDevelopper** [comment]
Ok got it, I had already Postman but I used it only one or two times. Then there are two JSON structure? One for GET and one for POST request?

**Comment by ghost** [comment]
Yes exactly. One is for login as u cannot get the request unless and until session is validated. So please do use that POST request - login api for logging in . by which you get the session_id . use this session_id value in the next GET request and get the json object. otherwise you would get an error. @iDevelopper  FYI, please check the screenshots .

**Comment by iDevelopper** [comment]
I wrote a sample to show how you could have a generic API Manager in your application (open the xcworkspace, because it use the SwiftyJSON pod):

[APIManagerSample.zip](https://github.com/John-Lluch/SWRevealViewController/files/1274103/APIManagerSample.zip)

**Comment by iDevelopper** [comment]
Now we have to add a parameter to the func data_request:

```
    func data_request(_ params: String, onSuccess: @escaping(JSON) -> Void, onFailure: @escaping(Error) -> Void)
        var urlString = baseURL + params
        urlString = urlString.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)!
...
```
Or add functions for different request types.

And create the different models to return these models instead a json.

What are the models you can receive?

**Comment by ghost** [comment]
hi @iDevelopper  Can you show me how do we pass values and populating Values Into tableview  as per the design standards following Design patterns from get request. Maybe i can reuse a single model as per the server architecture as all i do recieve are just string, int, double or boolean values.

**Comment by ghost** [comment]
[Demo.postman_collection.zip](https://github.com/John-Lluch/SWRevealViewController/files/1276233/Demo.postman_collection.zip)

FYI, these are the API's @iDevelopper

**Comment by iDevelopper** [comment]
So you have two types of request, ledger and community? Then we have to create two object class.

**Comment by iDevelopper** [comment]
Could we continue this discussion via mail as we are in an SWRevealViewController issue and it is not the right place for this! (patrick.bodet4@wanadoo.fr)?

**Comment by iDevelopper** [comment]
When you will send me a email, I'll send you a sample!

**Comment by ghost** [comment]
Hi Patrick,


                can you show me one ViewController  populating  with
Generic URL Request Handler so that i will follow the remaining
 ViewControllers. Please  i am new to swift thats why i am asking these
many no of times please Patrick. Please Guide me as a Teacher Please

**Comment by iDevelopper** [comment]
Yes I just want to continue elsewhere than here, because we are in a SWRevealViewController issue. Can I send you a sample at sanaboinaprasadofficial@gmail.com ?

**Comment by ghost** [comment]
Yes please do . anything is fine . I need the SWRevealController in the
third screen . So i think that wouldn't be an issue. But i need to follow
the MVC pattern w.r.t it .

**Comment by iDevelopper** [comment]
I sent you a sample yesterday. APIManagerSample. Did you receive it?

**Comment by ghost** [comment]
Oh yeah, I recieved . Sorry i missed out that email. Thank you.

**Comment by ghost** [comment]
Hi  Patrick,


Thanks for the help till now. I am very new to iOS Swift. I want to  code
like Professionals  with respective code optimisation. Can  you help me in
that.
